govcd/catalog_test.go

```diff
@@ -1307,6 +1307,161 @@ func (vcd *TestVCD) Test_CatalogAccessAsOrgUsers(check *C) {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	// Retrieve the shared catalog in the other organization
+	adminCatalog2, err := vcdClient2.Client.GetAdminCatalogById(adminCatalog1Created.AdminCatalog.ID)
+	check.Assert(err, IsNil)
+	check.Assert(adminCatalog2, NotNil)
+
+	// Retrieve the catalog from both tenants, using functions that don't rely on organization internals
+	catalog1FromOrg, err := vcdClient1.Client.GetCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, IsNil)
+	adminCatalog1FromOrg, err := vcdClient1.Client.GetAdminCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, IsNil)
+	catalog2FromOrg, err := vcdClient2.Client.GetCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, IsNil)
+	adminCatalog2FromOrg, err := vcdClient2.Client.GetAdminCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, IsNil)
+
+	// Also retrieve the catalog items from both tenants
+	vAppTemplate1, err := catalog1FromOrg.GetVAppTemplateByName(vappTemplateName)
+	check.Assert(err, IsNil)
+	check.Assert(vAppTemplate1.VAppTemplate.HREF, Equals, vAppTemplateAsSystem.VAppTemplate.HREF)
+	mediaRecord1, err := catalog1FromOrg.GetMediaByName(mediaName, false)
+	check.Assert(err, IsNil)
+	check.Assert(mediaRecord1.Media.HREF, Equals, mediaRecordAsSystem.Media.HREF)
+
+	vAppTemplate2, err := catalog2FromOrg.GetVAppTemplateByName(vappTemplateName)
+	check.Assert(err, IsNil)
+	check.Assert(vAppTemplate2.VAppTemplate.HREF, Equals, vAppTemplateAsSystem.VAppTemplate.HREF)
+	mediaRecord2, err := catalog2FromOrg.GetMediaByName(mediaName, false)
+	check.Assert(err, IsNil)
+	check.Assert(mediaRecord2.Media.HREF, Equals, mediaRecordAsSystem.Media.HREF)
+
+	check.Assert(catalog1FromOrg.Catalog.HREF, Equals, catalog1AsOrg1.Catalog.HREF)
+	check.Assert(adminCatalog1FromOrg.AdminCatalog.HREF, Equals, adminCatalog1Created.AdminCatalog.HREF)
+	check.Assert(adminCatalog2FromOrg.AdminCatalog.HREF, Equals, adminCatalog1Created.AdminCatalog.HREF)
+	check.Assert(catalog2FromOrg.Catalog.HREF, Equals, catalog1AsOrg1.Catalog.HREF)
+
+	isSharedReadOnly, err := adminCatalog1.IsSharedReadOnly()
+	check.Assert(err, IsNil)
+	check.Assert(isSharedReadOnly, Equals, true)
+
+	fmt.Println("removing read-only catalog sharing")
+	err = adminCatalog1Created.SetReadOnlyAccessControl(false)
+	check.Assert(err, IsNil)
+	catalog1FromOrg, err = vcdClient1.Client.GetCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, IsNil)
+	check.Assert(catalog1FromOrg, NotNil)
+	fmt.Println("try retrieving read-only catalog from second org")
+	time.Sleep(timeElapsedToAvailability)
+	adminCatalog2FromOrg, err = vcdClient2.Client.GetAdminCatalogByName(org1.AdminOrg.Name, catalogName)
+	check.Assert(err, NotNil)
+	check.Assert(adminCatalog2FromOrg, IsNil)
+
+	isSharedReadOnly, err = adminCatalog1.IsSharedReadOnly()
+	check.Assert(err, IsNil)
+	check.Assert(isSharedReadOnly, Equals, false)
+
+	timeout = 30 * time.Second
```

Comment:
What would happen if the deletion takes more than 30 seconds and there's a failure happening every time inside of the `for` loop? Would it be hard to check for errors after successful/unsuccessful deletion?

Reply:
Right. An assertion is missing after the loop. Thanks